cmake/options.cmake

```diff
@@ -0,0 +1,322 @@
+include(CMakeDependentOption)
+
+#------------------------------------------------------------------------------
+# Add an option with optional dependencies.
+#
+# Arguments:
+#   <DEFAULT_STATE> - Is the option enabled by default? (`ON` or `OFF`)
+#
+#   DEPENDS <expression>
+#     A list of expressions which must evaluate to true for the component to
+#     be made available. (Otherwise, the component will not be enabled, and the
+#     option for the component will be hidden.)
+#
+# Extra arguments are combined (with a single space) to form the description of
+# the option.
+#------------------------------------------------------------------------------
+function(drake_option NAME DEFAULT_STATE)
+  # "Fix" escaping so cmake_parse_arguments will split properly (note:
+  # cmake_parse_arguments splits twice, so escape two more levels).
+  string(REPLACE "\\" "\\\\\\" _args "${ARGN}")
+
```

Comment:
So, one answer to that question is that I kept adding more escaping until things worked :smile:. IOW, if one of those is omitted, it is experimentally known that splitting doesn't happen as desired.

In line with the comment... the input is escaped, the output must be escaped to avoid improper splitting, and `cmake_parse_arguments` will swallow two more levels (one when `${ARGN}` is expanded, and one internally). So the initial one level of escaping must be changed to 1+2=3 levels of escaping. Thus, six `\`s (to be seen as 3 after the text parsing pass).